geomesa-hbase/geomesa-hbase-security/src/main/java/org/locationtech/geomesa/hbase/security/HBaseAuthorizationsProvider.java

```diff
@@ -0,0 +1,34 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.hbase.security;
+
+
+import org.apache.hadoop.hbase.security.visibility.Authorizations;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public interface HBaseAuthorizationsProvider {
+    public static final String HBASE_AUTH_PROVIDER_SYS_PROPERTY = "geomesa.hbase.auth.provider.impl";
+
+    /**
+     * Gets the authorizations for the current context. This may change over time (e.g. in a multi-user environment),
+     * so the result should not be cached.
+     *
+     * @return
+     */
+    public Authorizations getAuthorizations();
```

Comment:
wonder if it would be worthwhile to abstract this out to a common interface that just returns a string or byte array, then have accumulo/hbase wrappers that create the appropriate class type. possibly then an auth implementation could be re-used between hbase and accumulo - although maybe that is not an actual use case